internal/types/compare.go

```diff
@@ -70,7 +111,7 @@ func Compare(docValue, filterValue any) CompareResult {
 				continue
 			}
 
-			if res := compareScalars(docValue, filterValue); res != Incomparable {
+			if res := compareScalars(docValue, filterValue); cond(res) {
```

Comment:
I didn't use `Max` or `Min`, this somehow got quite a bit complicated. For example comparing `$gte` with doc `bson.A{"c", "b", "a"}` with filter `bson.D{{"v", bson.D{{"$gte", 41.13}}}}`, I think it is greater because [string is greater than number](https://www.mongodb.com/docs/manual/reference/bson-type-comparison-order/#comparison-sort-order). But mongo DB doesn't think so according to the compat test. Anyway for now, let's skip max and min.

Reply:
I was totally wrong :see_no_evil:

Reply:
Okay, so it seems this way. If you have a docValue `bson.A{nil, "foo", 42}` and a filter `{"$gte", 43}`, compat returns less `<`. It seems to skip/ignores a string "foo" (which has [greater comparison order](https://www.mongodb.com/docs/manual/reference/bson-type-comparison-order/#comparison-sort-order)) but only compare against number 42. So getting max won't work because max is specific to type of value filter has.